float64 EstimateValue
bool Flag
bool EstFlag
